### PR TITLE
internal: ServerStatusParams should consider 'prime caches' in quiescent status

### PR DESCRIPTION
**Pull request by Wilfred**
Priming caches is a performance win, but it takes a lock on the salsa database and prevents rust-analyzer from responding to e.g. go-to-def requests.

This causes confusion for users, who see the spinner next to rust-analyzer in the VS Code footer stop, so they start attempting to navigate their code.

Instead, set the `quiescent` status in LSP to false during cache priming, so the VS Code spinner persists until we can respond to any LSP request.

### CODE REVIEW

**Comment by Wilfred** [comment]
Zulip thread: https://rust-lang.zulipchat.com/#narrow/stream/185405-t-compiler.2Frust-analyzer/topic/Making.20'prime.20caches'.20state.20explicit.20in.20VS.20Code.20UI

File: crates/rust-analyzer/src/reload.rs [hunk 1]
```diff
@@ -61,6 +61,10 @@ pub(crate) enum ProcMacroProgress {
 }
 
 impl GlobalState {
+    /// Is the server quiescent?
+    ///
+    /// This indicates that we've fully loaded the Cargo project and
+    /// can respond to (at least) some LSP requests.
```
Comment (Veykril) [inline crates/rust-analyzer/src/reload.rs:67]:
```suggestion
    /// This indicates that we've fully loaded the projects and
    /// are ready to do semantic work.
```

File: crates/rust-analyzer/src/reload.rs [hunk 2]
```diff
@@ -71,6 +75,15 @@ impl GlobalState {
             && self.vfs_progress_config_version >= self.vfs_config_version
     }
 
+    /// Is the server ready to respond to any LSP request?
```
Comment (Veykril) [inline crates/rust-analyzer/src/reload.rs:78]:
```suggestion
    /// Is the server ready to respond to analysis dependent LSP requests?
```

**Review by Veykril — APPROVED** [review]
Slight comment change suggestions

**Pull request merged**